classes/publish/app.js

```diff
@@ -106,7 +107,21 @@ module.exports = class PublishApp {
 
         this.log.debug('Loading import map file from server');
         try {
-            const maps = this.map.map(m => fetch(m).then(r => r.json()));
+            const maps = this.map.map(m =>
+                fetch(m).then(r => {
+                    if (r.status === 404) {
+                        throw new Error(
+                            'Import map could not be found on server',
+                        );
+                    } else if (r.status >= 400 && r.status < 500) {
+                        throw new Error('Server rejected client request');
+                    } else if (r.status >= 500) {
+                        throw new Error('Server error');
+                    } else {
+                        return r.json();
+                    }
```

Comment:
This `if/else` can actually be written as a switch statement. One can use expressions in the switch cases in js.

Reply:
like this?

```js
switch (true) {
    case r.status === 404:
        throw new Error(
            'Import map could not be found on server',
        );
    case r.status >= 400 && r.status < 500:
        throw new Error('Server rejected client request');
    case r.status >= 500:
        throw new Error('Server error');
    default:
        return r.json();
}
```

Is it any better you think? I suppose it might be a bit nicer